source/robot_model/include/robot_model/Model.hpp

```diff
@@ -736,4 +760,34 @@ inline std::vector<JointType> Model::get_joint_types() {
   }
   return joint_types;
 }
+
+inline Eigen::VectorXd
+Model::expand_joint_positions_to_full_configuration(const state_representation::JointPositions& joint_positions) const {
+  if (this->get_number_of_joints() == this->get_configuration_dimension()) {
+    return joint_positions.get_positions();
+  }
+  Eigen::VectorXd q(this->get_configuration_dimension());
+  const auto& joint_types = this->get_joint_types();
+  const Eigen::VectorXd& positions = joint_positions.get_positions();
+  std::size_t joint_input_idx = 0;
+  for (int i = 1; i < this->robot_model_.njoints; ++i) {
+    const auto& joint = this->robot_model_.joints[i];
+    std::size_t idx_q = joint.idx_q();
+    switch (joint_types[i - 1]) {
+      case JointType::CONTINUOUS: {
```

Comment:
Assuming we have a URDF with floating or planar joints, does it construct? Is there a chance we get to this switch case with that URDF?

Reply:
we should also perhaps consider a robot model factory that controllers/hardware can use, but that may imply we need to make a cleaner interface. That's why in a sense I was also considering

                RobotModel
               /                     \
RobotArmModel   MobileRobotModel

Reply:
A factory could also work even though I have a hard time seeing where we'd need that kind of inheritance since it feels like a controller is always just one or the other. But also an option

Reply:
It might actually also be unfavorable to derive a MobileRobotModel from the current RobotModel because one could maybe mess up things through pointer injection? Idk